### PR TITLE
refactor: regroup the paths into their own network.

### PR DESCRIPTION
**Pull request by hanwencheng**
This PR refactor the way to categorize accounts under their network, is the extend of #574, it address #573 , more specifically:

1. The network created by custom path should be categorized into their own network.
2. Now we support user created any path with any network, though it is not encouraged. For example, user should be able to create account under Westend network with path `//kusama//any_custom_path` and explicitly pointing that using westend network.

also rename "Custom Network" back to "Unknown Network", since now only account created in the previous version without appointing any network is listed here. (They currently use Kusama, but the hint text ask them to recreate it)

Previously:

| Kusama| Custom Network |
|----|----|
|Only contain account path prefixed with `//kusama`|contains all the custom path|
|IMG_0097|IMG_0098|

After this fix:
| Kusama | Unknown Network |
|---|---|
|Contain account with kusama network appointed and account prefixed with `//kusama`| contain only legacy account created without network appointed| 
|IMG_0096|IMG_0099|

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -165,9 +165,9 @@ android {
         release {
             // Caution! In production, you need to generate your own keystore file.
             // see https://facebook.github.io/react-native/docs/signed-apk-android.
-            // signingConfig signingConfigs.debug
-            // minifyEnabled enableProguardInReleaseBuilds
-            // proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
+//             signingConfig signingConfigs.debug
+             minifyEnabled enableProguardInReleaseBuilds
```
Comment (hanwencheng) [inline android/app/build.gradle]: These line helps to reduce the android package size

File: android/gradle.properties [hunk 2]
```diff
@@ -21,5 +21,6 @@ MYAPP_RELEASE_STORE_FILE=apk.keystore
 MYAPP_RELEASE_KEY_ALIAS=testing
 MYAPP_RELEASE_STORE_PASSWORD=testing
 MYAPP_RELEASE_KEY_PASSWORD=testing
+android.enableR8=false
```
Comment (hanwencheng) [inline android/gradle.properties]: use Proguard need to disable R8

File: src/types/networkSpecsTypes.ts [hunk 3]
```diff
@@ -47,9 +47,9 @@ export function isSubstrateNetworkParams(
 		| UnknownNetworkParams
 		| EthereumNetworkParams
 ): networkParams is SubstrateNetworkParams {
+	const { protocol, pathId } = networkParams as SubstrateNetworkParams;
 	return (
-		(networkParams as SubstrateNetworkParams).protocol ===
-		NetworkProtocols.SUBSTRATE
+		protocol === NetworkProtocols.SUBSTRATE && pathId !== unknownNetworkPathId
```
Comment (sjeohp) [inline src/types/networkSpecsTypes.ts]: This will also be true for `isUnknownNetworkParams` below.

File: src/types/networkSpecsTypes.ts [hunk 4]
```diff
@@ -71,8 +71,10 @@ export function isUnknownNetworkParams(
 		| UnknownNetworkParams
 		| EthereumNetworkParams
 ): networkParams is UnknownNetworkParams {
+	const { protocol, pathId } = networkParams as SubstrateNetworkParams;
 	return (
-		(networkParams as UnknownNetworkParams).protocol ===
-		NetworkProtocols.UNKNOWN
+		(protocol === NetworkProtocols.SUBSTRATE &&
+			pathId !== unknownNetworkPathId) ||
```
Comment (sjeohp) [inline src/types/networkSpecsTypes.ts]: This matches the same case as above.

Comment (sjeohp) [reply src/types/networkSpecsTypes.ts]:
```suggestion
		(isSubstrateNetworkParams(networkParams) ||
```

Comment (sjeohp) [reply src/types/networkSpecsTypes.ts]: The function name should be changed to `isSubstrateOrUnknownNetworkParams`, or else it should only be true for unknown.

File: src/components/PathCard.tsx [hunk 5]
```diff
@@ -133,8 +143,8 @@ export default function PathCard({
 		</View>
 	);
 
-	return network.protocol === NetworkProtocols.SUBSTRATE ||
-		network.protocol === NetworkProtocols.UNKNOWN
+	return isSubstrateNetworkParams(networkParams) ||
+		isUnknownNetworkParams(networkParams)
```
Comment (sjeohp) [inline src/components/PathCard.tsx]: This is equivalent to `return isUnknownNetworkParams(networkParams)`, see below.

File: src/screens/AccountNetworkChooser.tsx [hunk 6]
```diff
@@ -240,9 +241,15 @@ function AccountNetworkChooser({
 			}
 		} else {
 			const paths = Array.from(currentIdentity!.meta.keys());
-			if (isSubstrateNetworkParams(networkParams)) {
-				const listedPaths = getPathsWithSubstrateNetworkKey(paths, networkKey);
-				if (listedPaths.length === 0)
+			if (
+				isSubstrateNetworkParams(networkParams) ||
+				isUnknownNetworkParams(networkParams)
```
Comment (sjeohp) [inline src/screens/AccountNetworkChooser.tsx]: This is equivalent to `if (isUnknownNetworkParams(networkParams)`, see below.

File: src/types/networkSpecsTypes.ts [hunk 7]
```diff
@@ -71,8 +71,10 @@ export function isUnknownNetworkParams(
...
+		protocol === NetworkProtocols.UNKNOWN
```
Comment (sjeohp) [inline src/types/networkSpecsTypes.ts]:
```suggestion
		(protocol === NetworkProtocols.UNKNOWN
```

File: src/types/networkSpecsTypes.ts (hunk 4 above)
Comment (hanwencheng) [reply src/types/networkSpecsTypes.ts]: It is still unknown network, but need to change `pathId !== unknownNetworkPathId` to `pathId === unknownNetworkPathId`

**Pull request merged**